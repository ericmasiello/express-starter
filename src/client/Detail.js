/* @flow */
import React, { Component } from 'react';
import { Link } from 'react-router';
import './Detail.css';

// Note: Top level components must be class based for HMR to function
/* eslint-disable react/prefer-stateless-function */
export default class Detail extends Component {
  render() {
    return (
      <div className="detail">
        <h2>Detail page.</h2>
        <div>
          <Link to="/">Back Home</Link>
        </div>
      </div>
    );
  }
}
/* eslint-enable react/prefer-stateless-function */
