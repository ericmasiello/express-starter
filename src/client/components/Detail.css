.detail {
  color: darkblue;
  background-color: white;
}