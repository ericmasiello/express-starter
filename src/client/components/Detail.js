/* @flow */
import React, { Component } from 'react';
import { Link } from 'react-router';
import styles from './Detail.scss';
import type { Namespace$DetailParams } from '../../types';

// Note: Top level components must be class based for HMR to function
// eslint-disable-next-line react/prefer-stateless-function
export default class Detail extends Component {
  props: Namespace$DetailParams;

  render() {
    return (
      <div className={styles.detail}>
        <h2>Detail page for {this.props.params.id}.</h2>
        <div>
          <Link to="/">Back Home</Link>
        </div>
      </div>
    );
  }
}
