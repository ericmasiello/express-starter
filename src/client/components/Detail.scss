.detail {
  padding: 0 1rem;
}
