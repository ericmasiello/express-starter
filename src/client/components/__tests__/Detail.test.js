import React from 'react';
import renderer from 'react-test-renderer';
import Detail from '../Detail';

it('should render', () => {
  const props = {
    params: {
      id: '123',
    },
  };

  const component = renderer.create(
    <Detail {...props} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
