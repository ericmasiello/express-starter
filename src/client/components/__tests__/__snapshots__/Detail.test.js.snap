// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className={undefined}
>
  <h2>
    Detail page for 
    123
    .
  </h2>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Nemo consequuntur maxime in eaque, aut unde. Provident voluptate, recusandae inventore, mollitia, iusto ullam et obcaecati voluptatem accusantium minima adipisci, fugit sunt.
  </p>
</div>
`;
